kalamine/utils.py

```diff
@@ -35,8 +43,30 @@ class Layer(IntEnum):
     ALTGR = 4
     ALTGR_SHIFT = 5
 
+    def next(self) -> 'Layer':
+        """The next layer in the layer ordering."""
+        return Layer(int(self)+1)
+
+    def necromance(self) -> 'Layer':
+        """Remove the effect of the dead key if any."""
+        if self == Layer.ODK:
+            return Layer.BASE
+        elif self == Layer.ODK_SHIFT:
+            return Layer.SHIFT
+        return self
+
+@dataclass
+class DeadKeyDescr:
+    char: str
+    name: str
+    base: str
+    alt: str
+    alt_space: str
+    alt_self: str
+
+
+DEAD_KEYS = [DeadKeyDescr(**data) for data in load_data( "dead_keys.yaml")]
```

Comment:
OMFG I hope I can remember this `**` trick.

Reply:
That's one more reason why `dataclass` is nice. If your data easily fits into a json/yaml dict then they get very easy to (de)serialize.

Reply:
This becomes so expressive I want to cry. ^_^